pkg/scheduler/schedule_one.go

```diff
@@ -164,157 +223,59 @@ func (sched *Scheduler) schedulingCycle(ctx context.Context, state *framework.Cy
-	preBindStatus := fwk.RunPreBindPlugins(ctx, state, assumedPod, scheduleResult.SuggestedHost)
-	if !preBindStatus.IsSuccess() {
-		var reason string
-		if preBindStatus.IsUnschedulable() {
-			metrics.PodUnschedulable(fwk.ProfileName(), metrics.SinceInSeconds(start))
-			reason = v1.PodReasonUnschedulable
-		} else {
-			metrics.PodScheduleError(fwk.ProfileName(), metrics.SinceInSeconds(start))
-			reason = v1.PodReasonSchedulerError
-		}
-
-		// trigger un-reserve plugins to clean up state associated with the reserved Pod
-		fwk.RunReservePluginsUnreserve(ctx, state, assumedPod, scheduleResult.SuggestedHost)
-		if forgetErr := sched.Cache.ForgetPod(assumedPod); forgetErr != nil {
-			klog.ErrorS(forgetErr, "scheduler cache ForgetPod failed")
-		} else {
-			// "Forget"ing an assumed Pod in binding cycle should be treated as a PodDelete event,
-			// as the assumed Pod had occupied a certain amount of resources in scheduler cache.
-			// TODO(#103853): de-duplicate the logic.
-			if preBindStatus.IsUnschedulable() {
-				defer sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalqueue.AssignedPodDelete, func(pod *v1.Pod) bool {
-					return assumedPod.UID != pod.UID
-				})
-			} else {
-				sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalqueue.AssignedPodDelete, nil)
-			}
-		}
-		sched.FailureHandler(ctx, fwk, assumedPodInfo, preBindStatus.AsError(), reason, clearNominatedNode)
-		return
+	if status := fwk.RunPreBindPlugins(ctx, state, assumedPod, scheduleResult.SuggestedHost); !status.IsSuccess() {
+		return status
 	}
 
-	bindStatus := sched.bind(ctx, fwk, assumedPod, scheduleResult.SuggestedHost, state)
-	if !bindStatus.IsSuccess() {
-		var reason string
-		if bindStatus.IsUnschedulable() {
-			metrics.PodUnschedulable(fwk.ProfileName(), metrics.SinceInSeconds(start))
-			reason = v1.PodReasonUnschedulable
-		} else {
-			metrics.PodScheduleError(fwk.ProfileName(), metrics.SinceInSeconds(start))
-			reason = v1.PodReasonSchedulerError
-		}
-
-		// trigger un-reserve plugins to clean up state associated with the reserved Pod
-		fwk.RunReservePluginsUnreserve(ctx, state, assumedPod, scheduleResult.SuggestedHost)
-		if err := sched.Cache.ForgetPod(assumedPod); err != nil {
-			klog.ErrorS(err, "scheduler cache ForgetPod failed")
-		} else {
-			// "Forget"ing an assumed Pod in binding cycle should be treated as a PodDelete event,
-			// as the assumed Pod had occupied a certain amount of resources in scheduler cache.
-			// TODO(#103853): de-duplicate the logic.
-			if bindStatus.IsUnschedulable() {
-				defer sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalqueue.AssignedPodDelete, func(pod *v1.Pod) bool {
-					return assumedPod.UID != pod.UID
-				})
-			} else {
-				sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(internalqueue.AssignedPodDelete, nil)
-			}
-		}
-		sched.FailureHandler(ctx, fwk, assumedPodInfo, fmt.Errorf("binding rejected: %w", bindStatus.AsError()), reason, clearNominatedNode)
```

Comment:
Changed the returned error here, before `binding rejected: error...`, now `error...`. For the sake of logical unify.